editor/block-settings-menu/block-mode-toggle.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { IconButton } from '@wordpress/components';
+import { getBlockType } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { getBlockMode, getBlock } from '../selectors';
+import { toggleBlockMode } from '../actions';
+
+export function BlockModeToggle( { blockType, mode, onToggleMode } ) {
+	if ( blockType.supportHTML === false ) {
```

Comment:
Unsure how defensive we should be here, i.e. what if `blockType` is null, or in `getBlock( state, uid ).name` the property accesses on `null`? Not likely to ever occur, but would be destructive if it did.